libraries/Microsoft.Bot.StreamingExtensions/Payloads/Assemblers/PayloadAssembler.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.StreamingExtensions.Payloads
+{
+    internal abstract class PayloadAssembler
+    {
+        private object _syncLock = new object();
+
+        public PayloadAssembler(Guid id)
+        {
+            Id = id;
+            End = false;
+        }
+
+        public Guid Id { get; private set; }
+
+        public bool End { get; private set; }
+
+        protected static JsonSerializer Serializer { get; set; } = JsonSerializer.Create(SerializationSettings.DefaultSerializationSettings);
+
+        private Stream Stream { get; set; }
+
+        public Stream GetPayloadStream()
+        {
+            lock (_syncLock)
+            {
+                if (Stream == null)
+                {
+                    Stream = CreatePayloadStream();
+                }
+            }
+
+            return Stream;
+        }
+
+        public abstract Stream CreatePayloadStream();
+
```

Comment:
If the method is "CreatePayloadSTream", shouldn't the return type be "PayloadStream" ?